src/usePermission.ts

```diff
@@ -3,28 +3,31 @@ import { noop, off, on } from './misc/util';
 
 export type IState = PermissionState | '';
 
-interface IPushPermissionDescriptor extends PermissionDescriptor {
+interface ExtendedExperimentalPermissions extends Permissions {
+  query(permissionDesc: IPermissionDescriptor): ReturnType<Permissions['query']>;
+}
+
+interface ExperimentalPushPermissionDescriptor {
   name: 'push';
   userVisibleOnly?: boolean;
 }
 
-interface IMidiPermissionDescriptor extends PermissionDescriptor {
+interface ExperimentalMidiPermissionDescriptor {
   name: 'midi';
   sysex?: boolean;
 }
 
-interface IDevicePermissionDescriptor extends PermissionDescriptor {
+interface ExperimentalDevicePermissionDescriptor {
   name: 'camera' | 'microphone' | 'speaker';
   deviceId?: string;
 }
 
 export type IPermissionDescriptor =
```

Comment:
Sorry I must have confused your in https://github.com/streamich/react-use/issues/2092#issuecomment-910451191:

```suggestion
export type ExtendedPermissionDescriptor =
```

(because we are extending built-in `PermissionDescriptor`)

Reply:
Yeah I thought you were recommending to change the names back 👍